pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -311,13 +311,23 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	// nolint
 	switch externalSecret.Spec.Target.CreationPolicy {
-	case esv1beta1.CreatePolicyMerge:
-		err = patchSecret(ctx, r.Client, r.Scheme, secret, mutationFunc, externalSecret.Name)
 	case esv1beta1.CreatePolicyNone:
 		log.V(1).Info("secret creation skipped due to creationPolicy=None")
 		err = nil
+	case esv1beta1.CreatePolicyMerge:
+		err = patchSecret(ctx, r.Client, r.Scheme, secret, mutationFunc, externalSecret.Name)
 	default:
-		_, err = ctrl.CreateOrUpdate(ctx, r.Client, secret, mutationFunc)
+		op, opErr := ctrl.CreateOrUpdate(ctx, r.Client, secret, mutationFunc)
+		err = opErr
```

Comment:
If this error happen, aren't we masking it by overwriting it on line 324?

Reply:
Not actually, since `OperationResultCreated` and `err != nil` can never happen together.

If there is an error, then `op == OperationResultNone` (but that doesn't mean that always that `op == OperationResultNone` there is an error).